File: spring/mapstruct-spring-extensions-example/src/main/java/com/github/yukihane/spring/mapstructspringextensionsexample/CarMapper.java
package com.github.yukihane.spring.mapstructspringextensionsexample;

import org.mapstruct.Mapper;
import org.mapstruct.Mapping;
import org.mapstruct.Mappings;

@Mapper(uses = { WheelMapper.class, PedalMapper.class })
public interface CarMapper {
    @Mappings({
        @Mapping(source = "wheel", target = "steeringWheel"),
        @Mapping(source = "pedal", target = "footPedal"),
    })
    CarDto convert(Car car);
}

File: spring/mapstruct-spring-extensions-kotlin/src/main/kotlin/com/example/mapstruct_spring_extensions_kotlin/CarMapper.kt
package com.example.mapstruct_spring_extensions_kotlin

import org.mapstruct.Mapper
import org.mapstruct.Mapping
import org.mapstruct.Mappings


@Mapper(uses = [WheelMapper::class, PedalMapper::class])
interface CarMapper {
    @Mappings(Mapping(source = "wheel", target = "steeringWheel"), Mapping(source = "pedal", target = "footPedal"))
    fun convert(car: Car): CarDto
}
